p2p/p2p_test.go

```diff
@@ -48,407 +41,337 @@ func startNewService(t *testing.T, cfg *Config, msgSendChan chan []byte, msgRecC
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	bm := &BlockRequestMessage{
-		ID:            7,
+	// Create block request message (RequestedData: 1 = request header)
+	blockRequest := &BlockRequestMessage{
+		ID:            1,
 		RequestedData: 1,
 		StartingBlock: []byte{1, 1},
 		EndBlockHash:  optional.NewHash(true, endBlock),
 		Direction:     1,
 		Max:           optional.NewUint32(true, 1),
 	}
 
-	encMsg, err := bm.Encode()
+	encBlockRequest, err := blockRequest.Encode()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	err = sa.host.send(p, encMsg)
+	err = nodeA.host.send(*addrInfoB, encBlockRequest)
 	if err != nil {
-		t.Errorf("Send error: %s", err)
+		t.Fatal(err)
 	}
 
 	select {
-	case <-msgChan:
+	case message := <-msgSendB:
+		// Compare received message to original message
+		if !reflect.DeepEqual(message, encBlockRequest) {
+			t.Error("Did not receive the correct message")
+		}
 	case <-time.After(30 * time.Second):
-		t.Fatalf("Did not receive message from %s", sa.host.hostAddr)
+		t.Errorf("Did not receive message from %s", nodeA.host.hostAddr)
 	}
 }
 
 func TestGossiping(t *testing.T) {
-	//Start node A
-	nodeConfigA := &Config{
-		BootstrapNodes: nil,
-		Port:           7000,
-		NoBootstrap:    true,
-		NoMdns:         true,
-		RandSeed:       1,
+	configA := &Config{
+		Port:        7001,
+		RandSeed:    1,
+		NoBootstrap: true, // TODO: fix no bootstrap, this should be required
+		NoMdns:      true, // TODO: investigate failed dials, disable for now
 	}
 
-	nodeA := startNewService(t, nodeConfigA, nil, nil)
+	nodeA := startNewService(t, configA, nil, nil)
 	defer nodeA.Stop()
-	nodeAAddr := nodeA.host.fullAddrs()[0]
 
-	//Start node B
-	nodeConfigB := &Config{
-		BootstrapNodes: []string{
-			nodeAAddr.String(),
-		},
-		Port:     7001,
-		NoMdns:   true,
-		RandSeed: 2,
+	addrA := nodeA.host.fullAddrs()[0]
+
+	configB := &Config{
+		BootstrapNodes: []string{addrA.String()}, // Bootstrap node with node A
+		Port:           7002,
+		RandSeed:       2,
+		NoMdns:         true, // TODO: investigate failed dials, disable for now
 	}
 
-	msgChanB := make(chan []byte)
-	nodeB := startNewService(t, nodeConfigB, msgChanB, nil)
+	msgSendB := make(chan []byte)
+
+	nodeB := startNewService(t, configB, msgSendB, nil)
 	defer nodeB.Stop()
-	nodeBAddr := nodeB.host.fullAddrs()[0]
 
-	//Start node C
-	nodeConfigC := &Config{
-		BootstrapNodes: []string{
-			nodeBAddr.String(),
-		},
-		Port:     7002,
-		NoMdns:   true,
-		RandSeed: 3,
+	configC := &Config{
+		BootstrapNodes: []string{addrA.String()}, // Bootstrap node with node A
+		Port:           7003,
+		RandSeed:       3,
+		NoMdns:         true, // TODO: investigate failed dials, disable for now
 	}
 
-	msgChanC := make(chan []byte)
-	nodeC := startNewService(t, nodeConfigC, msgChanC, nil)
+	msgSendC := make(chan []byte)
+
+	nodeC := startNewService(t, configC, msgSendC, nil)
 	defer nodeC.Stop()
 
-	// Meaningless hash
+	// Create end block hash (hex has no significance here)
```

Comment:
I wouldn't say hex has no significance here, it's more like this hash is arbitrary

Reply:
Good point. :+1: